compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -1561,6 +1561,12 @@ impl<'a, 'tcx> InferCtxtPrivExt<'tcx> for InferCtxt<'a, 'tcx> {
                     let generics = self.tcx.generics_of(*def_id);
                     if generics.params.iter().any(|p| p.name != kw::SelfUpper)
                         && !snippet.ends_with('>')
+                        && ![
+                            self.tcx.lang_items().fn_trait(),
+                            self.tcx.lang_items().fn_mut_trait(),
+                            self.tcx.lang_items().fn_once_trait(),
+                        ]
+                        .contains(&Some(*def_id))
```

Comment:
I'm surprised there isn't a function for this. I guess there is
```suggestion
                        && self.tcx.fn_trait_kind_from_lang_item(*def_id).is_some()
```
which feels a little bit cleaner to me. But it looks like both are used in the compiler, and I don't have a strong opinion anyway.

Reply:
Changing it to use `fn_trait_kind_from_lang_item` feels appropriate.

Reply:
Fixed. This is ready for review again @estebank!